src/probnum/filtsmooth/statespace/integrator.py

```diff
@@ -0,0 +1,265 @@
+"""Integrated processes such as the integrated Wiener process or the Matern process.
+
+This is the sucessor of the former ODEPrior.
+"""
+try:
+    # cached_property is only available in Python >=3.8
+    from functools import cached_property
+except ImportError:
+    from cached_property import cached_property
+
+import numpy as np
+import scipy.special
+
+import probnum.random_variables as pnrv
+
+from . import discrete_transition, sde
+from .preconditioner import TaylorCoordinates
+
+
+class Integrator:
+    """
+    An integrator is a special kind of SDE,
+    where the :math:`i`th coordinate models the :math:`i`th derivative.
+    """
+
+    def __init__(self, ordint, spatialdim):
+        self.ordint = ordint
+        self.spatialdim = spatialdim
+
+    def proj2coord(self, coord: int) -> np.ndarray:
+        """
+        Projection matrix to :math:`i`-th coordinates.
+
+        Computes the matrix
+
+        .. math:: H_i = \\left[ I_d \\otimes e_i \\right] P^{-1},
+
+        where :math:`e_i` is the :math:`i`-th unit vector,
+        that projects to the :math:`i`-th coordinate of a vector.
+        If the ODE is multidimensional, it projects to **each** of the
+        :math:`i`-th coordinates of each ODE dimension.
+
+        Parameters
+        ----------
+        coord : int
+            Coordinate index :math:`i` which to project to.
+            Expected to be in range :math:`0 \\leq i \\leq q + 1`.
+
+        Returns
+        -------
+        np.ndarray, shape=(d, d*(q+1))
+            Projection matrix :math:`H_i`.
+        """
+        projvec1d = np.eye(self.ordint + 1)[:, coord]
+        projmat1d = projvec1d.reshape((1, self.ordint + 1))
+        projmat = np.kron(np.eye(self.spatialdim), projmat1d)
+        return projmat
+
+
+class IBM(Integrator, sde.LTISDE):
+    """Integrated Brownian motion in :math:`d` dimensions."""
+
+    def __init__(self, ordint, spatialdim, diffconst):
+        self.diffconst = diffconst
+
+        # initialise BOTH superclasses' inits.
+        # I don't like it either, but it does the job.
```

Comment:
any idea how to better initialise this multiple inheritance thing here? the code works, is kind of readable, but I am not sure how "best practice" this is.

Reply:
I don't know either. To be honest, I don't even know what `super()` does in this case^^

Edit: I tried it out. It seems you could fix that to a single call of `super().__init__()`: https://stackoverflow.com/a/16310777

Reply:
Oh but I don't know if there is a clean way of handling the different arguments. So all in all I guess what you're doing here is fine :)

Reply:
Yeah according to StackOverflow you can do this in a single super(), though in my code this never worked^^ The present thing worked though, which is why it is there :D